### DISCUSSION TITLE
homebrew-core unshallow ends with error

### DISCUSSION THREAD

**Discussion by rusho** [Homebrew/discussions#355] [Everyday usage] [answered]
Hello I have a strange issue when I try to unshallow `homebrew-core`: 

```
rusho:~/ $ git -C /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core fetch --unshallow             
remote: Enumerating objects: 251650, done.
remote: Counting objects: 100% (248461/248461), done.
remote: Compressing objects: 100% (117525/117525), done.
error: --shallow-file died of signal 25, 47.43 MiB | 4.53 MiB/s  
fatal: index-pack failed
```

`homebrew-cask` proceeded just fine but due to inability do do the same on core, I'm unable to `brew update` 😒 Any idea what's up? Or how could I reinstall homebrew without loosing everything I already installed via `brew` (afaik standard uninstall script will remove also all stuff I have installed via `brew`, please correct me if I'm wrong)

**Comment by carlocab** [comment]
Do you have changes to `$(brew --repo homebrew/core)` that you're not willing to throw away? If you don't have any such changes, try:
```
rm -rf /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
brew tap homebrew/core
brew update
```
If tapping doesn't work for some reason, you can always
```
cd /usr/local/Homebrew/Library/Taps/homebrew
git clone https://github.com/Homebrew/homebrew-core.git
```

**Comment by gromgit** [comment, marked as answer]
Signal 25 (`SIGXFSZ`) means you've exceeded your file size resource limit, which is really weird. What's the output of `ulimit -a`, when run in the same shell as your unshallow attempt?

**Reply by rusho** [reply]
> Signal 25 (`SIGXFSZ`) means you've exceeded your file size resource limit, which is really weird. What's the output of `ulimit -a`, when run in the same shell as your unshallow attempt?

I have this (not sure if it's enough, I know since Mojave or Catalina updating these on MacOS started to be a pain in the ass):

```
rusho:~/ $ ulimit -a                                                                                                                                            
-t: cpu time (seconds)              unlimited
-f: file size (blocks)              100000
-d: data seg size (kbytes)          unlimited
-s: stack size (kbytes)             8192
-c: core file size (blocks)         0
-v: address space (kbytes)          unlimited
-l: locked-in-memory size (kbytes)  unlimited
-u: processes                       4176
-n: file descriptors                100000
```

**Reply by gromgit** [reply]
> -f: file size (blocks)              100000

This means any single file written by any descendant process of your shell can only write up to 100,000 * 512 ~ **51MB**. The core tap payload that `git` is pulling from GitHub is definitely larger than that, so this error is to be expected:
```
error: --shallow-file died of signal 25, 47.43 MiB | 4.53 MiB/s  
```

Run `ulimit -f unlimited`, then do your unshallow again.

**Reply by rusho** [reply]
yeah I bumped that value and it works, thanks. Weird though, I use to work with much bigger repositories/payloads and I never used to have problems before, I guess Big Sur upgrade must have messed up something. Thanks for help!
